File: warehouse/src/main/java/sep3/warehouse/DTO/products/CreateProductDto.java
package sep3.warehouse.DTO.products;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import sep3.warehouse.DTO.brands.BrandDTO;
import sep3.warehouse.entities.Product;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CreateProductDto {
    private String name;
    private String description;
    private double price;
    private String imagePath;
    private BrandDTO brand;

    public static Product mapCreateProductDtoToProduct(CreateProductDto createProductDto) {
        Product product = new Product();
        product.setName(createProductDto.getName());
        product.setDescription(createProductDto.getDescription());
        product.setPrice(createProductDto.getPrice());
        product.setImagePath(createProductDto.getImagePath());
        product.setBrand(BrandDTO.mapBrandDtoToBrand(createProductDto.getBrand()));

        return product;
    }
}

File: DTO/Product/Create/CreateProductDTO.cs
namespace sep3.DTO.Product.Create;

public class CreateProductDTO
{
    public string Name { get; set; }
    public string Description { get; set; }
    public double Price { get; set; }
    public string ImagePath { get; set; }
    public BrandDTO Brand { get; set; }
    
    public override string ToString()
    {
        return $"Name: {Name}, Description: {Description}, Price: {Price}, ImagePath: {ImagePath}, Brand: {Brand?.Name}";
    }
}